### ISSUE TITLE
How can I do refresh in electron html app 

### ISSUE THREAD

**Issue by johnico** [electron/electron#11895]
How can I do refresh in electron html app ?  I have window without menu and I wan to let user to refresh the page with F5 or control +R

* Electron version: 1.8.2
* Operating system: windows 7

### Expected behavior

F5 to refresh the page  or control+R

### Actual behavior

nothing happen

### How to reproduce
here is my main.js :
```
const electron = require('electron')
// Module to control application life.
const app = electron.app
// Module to create native browser window.
const BrowserWindow = electron.BrowserWindow

const path = require('path')
const url = require('url')

// Keep a global reference of the window object, if you don't, the window will
// be closed automatically when the JavaScript object is garbage collected.
let mainWindow

function createWindow () {
  // Create the browser window.
  mainWindow = new BrowserWindow({
		webPreferences:{
			nodeIntegration:false
		},
		width: 1920,
		height: 1080,
		options:{
			fullscreen:true
		}

		
	})

  // and load the index.html of the app.
  mainWindow.loadURL(url.format({
    pathname: path.join(__dirname, 'index.html'),
    protocol: 'file:',
    slashes: true
  }))

  // Open the DevTools.
  // mainWindow.webContents.openDevTools()

  // Emitted when the window is closed.
  mainWindow.on('closed', function () {
    // Dereference the window object, usually you would store windows
    // in an array if your app supports multi windows, this is the time
    // when you should delete the corresponding element.
    mainWindow = null
  })
  
  mainWindow.setMenu(null);

  
  mainWindow.setFullScreen(true);
  mainWindow.maximize();
	
  
}

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.on('ready', createWindow)

// Quit when all windows are closed.
app.on('window-all-closed', function () {
  // On OS X it is common for applications and their menu bar
  // to stay active until the user quits explicitly with Cmd + Q
  if (process.platform !== 'darwin') {
    app.quit()
  }
})

app.on('activate', function () {
  // On OS X it's common to re-create a window in the app when the
  // dock icon is clicked and there are no other windows open.
  if (mainWindow === null) {
    createWindow()
  }
})

// In this file you can include the rest of your app's specific main process
// code. You can also put them in separate files and require them here.

```

-->

**Comment by TiagoDanin** [comment]
Import and use the [globalShortcut](http://electronjs.org/docs/api/global-shortcut)
And use the `mainWindow.reload()`

---------

Example:
```javascript
//***//
const globalShortcut = electron.globalShortcut
//***//
function createWindow () {
	//***//
	globalShortcut.register('f5', function() {
		console.log('f5 is pressed')
		mainWindow.reload()
	})
	globalShortcut.register('CommandOrControl+R', function() {
		console.log('CommandOrControl+R is pressed')
		mainWindow.reload()
	})
}
```

**Comment by danyadev** [comment]
or in renderer process use
`
const {getCurrentWindow, globalShortcut} = require('electron').remote;
globalShortcut.register('f5', function() {
		console.log('f5 is pressed')
		getCurrentWindow().reload()
})
`

**Comment by MarshallOfSound** [comment]
GitHub issues are for feature requests and bug reports, questions about using Electron or code assistance requests should be directed to the [community](https://github.com/electron/electron#community) or to the [Slack Channel](http://atom-slack.herokuapp.com/).

**Comment by ThunbergOlle** [comment]
If I want to reload the mainWindow from another windows remote, how do I do that?

**Comment by randonia** [comment]
I know this is closed but posting here for posterity, and to clarify my 👎 reactions - 

`globalShortcut` will register your shortcut **globally** with the OS, which means _no matter where you press that keybind you will trigger your event_

[from the documentation](https://electronjs.org/docs/api/global-shortcut#globalshortcut):

> The globalShortcut module can register/unregister a global keyboard shortcut with the operating system so that you can customize the operations for various shortcuts.
> 
> Note: The shortcut is global; it will work even if the app does not have the keyboard focus

**Comment by VivekNair** [comment]
Is there another strategy that I can employ to prevent @randonia 's issue? I only want to refresh is the window is in active focus.

**Comment by paulfwood** [comment]
The best solution I found was in the Electron [keyboard shortcuts](https://electronjs.org/docs/tutorial/keyboard-shortcuts?query=Cmd%2BR#shortcuts-within-a-browserwindow) documentation. It suggests using [mousetrap](https://github.com/ccampbell/mousetrap) which doesn't create global key triggers, allows multiple shortcuts, and also allows you to create single key shortcuts, such that 'c' can open the 'create new email' window for example.

This is the code to refresh the window:

	Mousetrap.bind(['command+r', 'ctrl+r', 'f5'], () => {
		mainWindow.reload()

		return false // prevents default behavior and stops event from bubbling
	})

**Comment by PByH** [comment]
Hi everyone!
I also faced with the same problem while debugging on macOS via chrome remote desktop (the command button is not working).
Here is my solution:
![111](https://user-images.githubusercontent.com/25103128/80458406-ecbb9600-895a-11ea-85f4-81671fb089b2.jpg)

**Comment by oguimbal** [comment]
@danyadev solution is OK, but it catches every further CTRL+R, even when the Electron window has no focus (at least on my machine), which is quite annoying.

Here is a version that fixes it:

```js
const {getCurrentWindow, globalShortcut} = require('electron').remote;

const reload = ()=> getCurrentWindow().reload();
const unregister = () => globalShortcut.unregister('CommandOrControl+R', reload);
const register = () => globalShortcut.register('CommandOrControl+R', reload); // or 'F5';
register();
window.addEventListener('focus', register);
window.addEventListener('blur', unregister);
window.addEventListener('beforeunload', unregister);
```

**Comment by jadrake75** [comment]
@oguimbal solution worked, however some times if I ALT-TAB or other action from the electron frame it was not unregistering right away.  I found adding

`getCurrentWindow().on('blur', unregister);`

aided to avoid this consistently.  The difference being of course a blur of the web frame vs. the electron application.

**Comment by BartenderWinery** [comment]
> or in renderer process use
> 
> ```js
> const {getCurrentWindow, globalShortcut} = require('electron').remote;
> 
> var reload = ()=>{
>   getCurrentWindow().reload()
> }
> 
> globalShortcut.register('F5', reload);
> globalShortcut.register('CommandOrControl+R', reload);
> // here is the fix bug #3778, if you know alternative ways, please write them
> window.addEventListener('beforeunload', ()=>{
>   globalShortcut.unregister('F5', reload);
>   globalShortcut.unregister('CommandOrControl+R', reload);
> })
> ```

Remote module is a horrible idea, this can give anyone with successful cross-site attacks full access to your computer's shell.
